taipy/gui/gui.py

```diff
@@ -1052,7 +1051,11 @@ def __send_var_list_update(  # noqa C901
                 modified_vars.remove(k)
         for _var in modified_vars:
             newvalue = values.get(_var)
-            if isinstance(newvalue, _TaipyData):
+            resource_handler = get_current_resource_handler()
+            custom_page_filtered_types = resource_handler.data_layer_supported_types if resource_handler else ()
+            if isinstance(newvalue, (_TaipyData)) or (
+                custom_page_filtered_types and isinstance(newvalue, custom_page_filtered_types)
+            ):
                 # A changing integer that triggers a data request
                 # newvalue = Gui._data_request_counter
```

Comment:
change the comment and remove the dead code

Reply:
I will remove it! I thought that you want to keep it for some reason :D